src/app/git-hub-organizations.service.ts

```diff
@@ -32,36 +35,60 @@ export class GitHubOrganizationsService {
+
+        } else {
+            console.log("=====> not cached :(");
+
+            // Check page number: it should
+            if (page < 1 || page > this.pageMarkers.length) {
+                const errorMessage = `Invalid page number: ${page}, should be between 1 and ${this.pageMarkers.length}`;
+                return new Promise((resolve, reject) => reject(new RangeError(errorMessage)));
+            }
+
+            // If this is the first call, ignore the page number and get the first page
+            if (this.pageMarkers.length === 1) {
+                page = 1;
+            }
+
+            // Build the organization page URL, see https://docs.github.com/en/rest/reference/orgs#list-organizations
+            // Need to retrieve entries whose ID is larger than the largest ID of the previous page
+            const previousPage = page - 1;
+            const previousPageMarker = this.pageMarkers[previousPage];
+            const url = this.organizationsUrl + pageSize + '&since=' + previousPageMarker;
+
+            // Read the requested page and, before we return the promise, save
+            // the ID of the last organization in the pageMarkers array
+            const promise = this.httpClient.get<GitHubOrganization[]>(url).toPromise();
+            promise.then((organizationPage) => {
+                const actualPageSize = organizationPage.length;
+                const lastOrganizationIndex = actualPageSize - 1;
+                const lastOrganization = organizationPage[lastOrganizationIndex];
+                this.pageMarkers[page] = lastOrganization.id;
+            });
+            let cacheData = new CacheData();
+            cacheData.pageSize = pageSize;
+            cacheData.page = page;
+            cacheData.data = promise;
+            this.cache.push(cacheData);
+            return promise;
         }
+    }
 
-        // Build the organization page URL, see https://docs.github.com/en/rest/reference/orgs#list-organizations
-        // Need to retrieve entries whose ID is larger than the largest ID of the previous page
-        const previousPage = page - 1;
-        const previousPageMarker = this.pageMarkers[ previousPage ];
-        const url = this.organizationsUrl + pageSize + '&since=' + previousPageMarker;
-
-        // Read the requested page and, before we return the promise, save
-        // the ID of the last organization in the pageMarkers array
-        const promise = this.httpClient.get<GitHubOrganization[]>( url ).toPromise();
-        promise.then( (organizationPage) => {
-            const actualPageSize = organizationPage.length;
-            const lastOrganizationIndex = actualPageSize - 1;
-            const lastOrganization = organizationPage[ lastOrganizationIndex ];
-            this.pageMarkers[ page ] = lastOrganization.id;
-        });
-
-        return promise;
+    getCachedData(pageSize: number, page: number): Promise <void | GitHubOrganization[]> {
+        for (let c of this.cache){
```

Comment:
Minor comment: here is no need to loop over the whole array here. Since you are always appending to the end of the array (with `cache.push()`) you can find if a page is caches simply by comparing the page index (`number`) with the array size:
```
if( number < this.cache.length ) { 
    ...
}
```